src/server_manager/web_app/app.ts

```diff
@@ -200,136 +209,218 @@ export class App {
       this.appRoot.openGetConnectedDialog(this.getS3InviteUrl(event.detail.accessKey, true));
     });
 
+    appRoot.addEventListener('ShowServerRequested', (event: PolymerEvent) => {
+      const displayServerId = event.detail.displayServerId;
+      const serverEntry = this.serverDisplayTable[displayServerId];
+      if (!serverEntry && !!this.serverBeingCreated) {
+        this.showServerCreationProgress();
+      } else if (!!serverEntry) {
+        this.showServerIfHealthy(serverEntry.server, serverEntry.displayServer);
+      } else {
+        // We should never reach this if the server display table is synced.
+        console.error(`Could not find server entry for display server ID ${displayServerId}`);
+      }
+    });
+
     onUpdateDownloaded(this.displayAppUpdateNotification.bind(this));
   }
 
   start(): void {
-    // Load manual servers from storage.
-    this.manualServerRepository.listServers().then((manualServers) => {
-      // Show any manual servers if they exist.
-      if (manualServers.length > 0) {
-        this.showManualServerIfHealthy(manualServers[0]);
-        return;
-      }
+    // Load display servers and associate them with managed and manual servers.
+    this.displayServerRepository.listServers().then((displayServers) => {
+      this.appRoot.serverList = displayServers;
+
+      this.manualServerRepository.listServers().then((manualServers) => {
+        this.syncServersToDisplay(manualServers).then(() => {
+          this.maybeShowLastDisplayedServer();
+        });
+      });
 
-      // User has no manual servers - check if they are logged into DigitalOcean.
       const accessToken = this.digitalOceanTokenManager.getStoredToken();
       if (accessToken) {
-        this.enterDigitalOceanMode(accessToken);
-        return;
+        this.enterDigitalOceanMode(accessToken).then((managedServers) => {
+          this.syncServersToDisplay(managedServers).then(() => {
+            this.maybeShowLastDisplayedServer();
+          });
+        });
       }
+    });
+    this.showIntro();
+  }
 
-      // User has no manual servers or DigitalOcean token.
-      this.showIntro();
+  private async syncServersToDisplay(servers: server.Server[]) {
+    for (const server of servers) {
+      await this.syncServerToDisplay(server);
```

Comment:
Why wait - won't this only slow things down?

Reply:
Actually, the wait is needed to ensure that the UI server list is synced after calling this method.

Reply:
OK. That's worth a comment. Seems like this function could use `Promise.all` to run the queries in parallel. But not a big deal.

Reply:
What I don't like about `Promise.all` is that if a single promise rejects, it rejects. What I have learned from using async/await is that await should always be called to ensure deterministic results. I'll merge now and add the comment in a follow up PR, if that's okay with you.